pkg/nvml/lib.go

```diff
@@ -282,20 +276,23 @@ func WithLibraryPath(path string) LibraryOption {
 func SetLibraryOptions(opts ...LibraryOption) error {
 	libnvml.Lock()
 	defer libnvml.Unlock()
-	if libnvml.dl != nil {
+	if libnvml.refcount != 0 {
 		return errLibraryAlreadyLoaded
 	}
 
+	o := libraryOptions{}
 	for _, opt := range opts {
-		opt(&libnvml)
+		opt(&o)
 	}
 
-	if libnvml.path == "" {
-		libnvml.path = defaultNvmlLibraryName
+	if o.path == "" {
+		o.path = defaultNvmlLibraryName
 	}
-	if libnvml.flags == 0 {
-		libnvml.flags = defaultNvmlLibraryLoadFlags
+	if o.flags == 0 {
+		o.flags = defaultNvmlLibraryLoadFlags
 	}
 
+	libnvml.dl = dl.New(o.path, o.flags)
```

Comment:
Note that `libnvml` still has a `path` member. Should we set that to `o.path` here too?

Reply:
Updated